docker/CMSRucioClient/scripts/syncUserRoles.py

```diff
@@ -0,0 +1,254 @@
+    egroup_map = defaultdict(set)
+    for _, values in all_groups.items():
+        if values['tag_relation'] == tag_relation and values['role'] == role:
+            tag_name = values['tag_name'].lower()
+            for egroup in values['egroups']:
+                egroup_map[tag_name].add(egroup)
+
+    return egroup_map
+
+
+#############################################
+# Group identities sync
+############################################
+
+def sync_roles_to_groups(client, users):
+    """
+    Sync roles to respctive group accounts
+    - local_users: site groups
+    - group_users: physics groups
+    """
+
+    local_users = defaultdict(set)
+    group_users = defaultdict(set)
+
+    # Both can be used as one as well
+    dn_email_map_local = {}
+    dn_email_map_group = {}
+
+    site_map = build_site_facility_map()
+    all_groups = make_request(CRIC_GROUP_API)
+    # get groups for group email accounts
+    facility_egroups = get_egroup_map(
+        all_groups, tag_relation="facility", role="local-data-manager")
+    group_egroups = get_egroup_map(
+        all_groups, tag_relation="group", role="group-data-manager")
+
+    for user in users:
+        roles = user['ROLES']
+        dn = user['DN']
+        email = user['EMAIL']
+
+        # data-managers: are site managers / admins?
+        # We also add them to list of local users
+        if 'data-manager' in roles:
+            for tag in roles['data-manager']:
+                if tag.startswith('site:'):
+                    site = tag.replace('site:', '', 1)
+                    local_users[site].add(dn)
+                    dn_email_map_local[dn] = email
+
+        # local-data-managers: set of users that manage data for #TODO (the site)?
+        if 'local-data-manager' in roles:
+            for tag in roles['local-data-manager']:
+                if tag.startswith('site:'):
+                    site = tag.replace('site:', '', 1)
+                    local_users[site].add(dn)
+                    dn_email_map_local[dn] = email
+
+        # group-data-manager: set of users that manage data for a physics group
+        if 'group-data-manager' in roles:
+            for tag in roles['group-data-manager']:
+                if tag.startswith('group:'):
+                    group = tag.replace("group:", '', 1)
+                    group_users[group].add(dn)
+                    dn_email_map_group[dn] = email
+
+    for site, dns in local_users.items():
+        try:
+            # At this point site-map does not contain deleted/disabled sites
+            # (i.e it only contains site with state=ACTIVE) this is the default result from the CRIC api
+            # These sites-tags are however still present under users
+            # TODO: Should we include / exclude them?
+            if site in site_map:
+                facility = site_map[site]
+                egroup = facility_egroups[facility]
+                set_identities(client=client, account=site, egroup=egroup,
```

Comment:
I found that the rucio account is actually not the same as the site, in contrast to group account names. 
So need to correct this. I will change it to the logic from here, https://github.com/dmwm/CMSRucio/blob/46b5e03cdaa9d2f7fb58b68fc95ed4828847122d/docker/CMSRucioClient/scripts/syncRolesToAttributesLocalUsers.py#L63-L66

Reply:
I was wondering that so that we can avoid collisions. I guess _group on the end could be good. Too bad we can't programmatically tell if it's a physics or detector or physics-object group.

Reply:
The 3 group accounts other than `local_users` in rucio are created under the `data-manager` role instead of `group-data-manager` as in the case of the `pog_tau` account. 

Do you think we can make use of this, somehow?

heavy-ions cric: https://cms-cric.cern.ch/api/accounts/group/query/list/?json&tag_name=heavy-ions
pog_tau cric: https://cms-cric.cern.ch/api/accounts/group/query/list/?json&tag_name=pog_tau

Reply:
I don't think it does, unfortunately. I think if we add _group to the end and translate dash to underscore that's good.